server/internal/core/application/covenantless.go

```diff
@@ -1501,6 +1502,74 @@ func (s *covenantlessService) finalizeRound(notes []note.Note, roundEndTime time
 	log.Debugf("finalized round %s with round tx %s", round.Id, round.Txid)
 }
 
+func (s *covenantlessService) waitForForfeitsAndBoardingSigs(
+	ctx context.Context,
+	remainingTime time.Duration,
+	numOfBoardingInputs int,
+) error {
+	waitCtx, cancel := context.WithTimeout(ctx, remainingTime)
+	defer cancel()
+
+	doneCh := make(chan struct{}, 1)
+	errCh := make(chan error, 1)
+
+	go func() {
+		ticker := time.NewTicker(time.Second / 3)
+		defer ticker.Stop()
+
+		for {
+			select {
+			case <-waitCtx.Done():
```

Comment:
we have 2 functions listining to the same channel.. this means only one of them will receive the done message from waitCtx. We can't control whom, so this can lead to unexpected behaviors.

In general i'm not sure about this approach, considering that clients submit sigs for their boarding inputs by calling SubmitSignedForfeitTxs.

I'd like to stick with the existing one and fixi where we signal that forfeit txs collection ended. At the moment we likely wait only for forfeits, we should wait also for boarding inputs.

@louisinger can you give more insights please?